test/test_course.py

```diff
@@ -100,7 +114,32 @@ def test_find_many_students_duplicate_students_returns_first_occurrence_index(se
         course.add(Student("Niles", "MacDonald", 192837465))
         course.add(Student("Jane", "Doe", 987654321))
         student_there = Student("Jane", "Doe", 987654321)
-        self.assertEqual(1, course.find(student_there))
+        self.assertEqual(1, course._find(student_there))
+
+    def test_contains_many_students_student_does_not_exist_returns_false(self):
```

Comment:
Why are these separated from the other `contains` tests?

Reply:
I see. I tend group units under test together, instead of the condition under test. That's probably opinion, I find it's a more natural grouping.